docs/diagrams/FavActivityDiagram.puml

```diff
@@ -0,0 +1,23 @@
+@startuml
+skin rose
+skinparam ActivityFontSize 15
+skinparam ArrowFontSize 12
+start
+:User executes fav command;
+
+'Since the beta syntax does not support placing the condition outside the
+'diamond we place it as the true branch instead.
+
+if () then ([index is valid])
+    if () then ([Person has favourite as false])
+        :Construct a new Person object with favourite as true;
+        :Replace the original Person with the new one;
+        :Refresh the list of Person;
+    else ([else])
+        :Throws CommandException that Person has favourite as true;
+    endif
+else ([else])
+    :Throw CommandException that index is is invalid;
```

Comment:
I think activity diagrams are not meant to have technical terms e.g `Throws CommandException`, perhaps `Show error message to user that person is already a favourite` is better?
on the same note,
- the "if" condition may be better phrased as `Person has favourite as false` -> `Person is favourite`
- line 13 `Construct a new Person object with favourite as true` -> `Create new Person with same values but is a favourite`
- line 17 `Person has favourite as true` -> `Person is already a favourite`

nit: line 15 `list of Person` -> `list of Persons`
nit: line 20 `index is is invalid` -> `index is invalid`

Reply:
I understand, thank you for pointing it out! With that said, I think `EditActivityDiagram` also contains those errors.

Reply:
oh, will have to correct that in another PR, thanks for pointing out!